mm2src/coins/nft.rs

```diff
@@ -1396,3 +1372,44 @@ pub(crate) fn get_domain_from_url(url: Option<&str>) -> Option<String> {
     url.and_then(|uri| Url::parse(uri).ok())
         .and_then(|url| url.domain().map(String::from))
 }
+
+/// Clears NFT data from the database for specified chains.
+pub async fn clear_nft_db(ctx: MmArc, req: ClearNftDbReq) -> MmResult<(), ClearNftDbError> {
+    if req.chains.is_none() && !req.clear_all {
+        return MmError::err(ClearNftDbError::InvalidRequest(
+            "Nothing to clear was specified".to_string(),
+        ));
+    }
+    let nft_ctx = NftCtx::from_ctx(&ctx).map_to_mm(ClearNftDbError::Internal)?;
+    let storage = nft_ctx.lock_db().await?;
```

Comment:
nit suggestion: I believe we can duplicate nft_ctx and storage within the if/else statements to avoid the need for checking `req` multiple times (both at the beginning of the function and below)

Reply:
changed the code to check req params once, also `nft_ctx` and `storage` are created only when necessary